### PR TITLE
concurrency_thread_synchronization / it-academy / JC1-117-22 / Victor Polonnikov / Done tasks for 2 projects

### PR DESCRIPTION
**Pull request by VictPol**
## Pull request state

Actual status for PR:
- [x] implemented
- [x] reviewed
- [ ] fixed
- [ ] approved

### CODE REVIEW

File: src/main/java/com/rakovets/course/java/core/practice/concurrency_thread_synchronization/project_producer_consumer/task_05/ConsumerThread.java [hunk 1]
```diff
@@ -0,0 +1,27 @@
+package com.rakovets.course.java.core.practice.concurrency_thread_synchronization.project_producer_consumer.task_05;
+
+import com.rakovets.course.java.core.practice.concurrency_thread_synchronization.project_producer_consumer.task_03.Store;
+
+import java.util.logging.Logger;
+
+public class ConsumerThread extends Thread {
```
Comment (rakovets) [inline src/main/java/com/rakovets/course/java/core/practice/concurrency_thread_synchronization/project_producer_consumer/task_05/ConsumerThread.java:7]: move to one package all tasks

File: src/main/java/com/rakovets/course/java/core/practice/concurrency_thread_synchronization/project_sky_net/task_01/Demo.java [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+    public static void main(String[] args) {
+        final int TIME_OF_WORK_IN_MINUTES = 10;
+
+        FactoryStore factoryStore = new FactoryStore(TIME_OF_WORK_IN_MINUTES);
+        Thread factoryStarter = new FactoryStarter("FactoryStarter", factoryStore);
+        Thread fractionOfRobotsStarter = new FractionOfRobotsStarter("FractionOfRobotsStarter", factoryStore);
+        factoryStarter.start();
+        fractionOfRobotsStarter.start();
+    }
+}
```
Comment (rakovets) [inline src/main/java/com/rakovets/course/java/core/practice/concurrency_thread_synchronization/project_sky_net/task_01/Demo.java:13]: over-engineering

File: src/main/java/com/rakovets/course/java/core/practice/concurrency_thread_synchronization/project_producer_consumer/task_02/Store.java [hunk 3]
```diff
@@ -0,0 +1,25 @@
+package com.rakovets.course.java.core.practice.concurrency_thread_synchronization.project_producer_consumer.task_02;
+
+import java.util.Deque;
+import java.util.logging.Logger;
+
+public class Store {
+    Logger logger = Logger.getLogger(Store.class.getName());
+
+    private Deque<Integer> deque;
+    public Store(Deque<Integer> deque) {
```
[the remark this answers is not in the document]
Comment (VictPol) [reply src/main/java/com/rakovets/course/java/core/practice/concurrency_thread_synchronization/project_producer_consumer/task_02/Store.java:10]: fixed

File: src/main/java/com/rakovets/course/java/core/practice/concurrency_thread_synchronization/project_producer_consumer/task_02/Store.java [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+        this.deque = deque;
+    }
+
+    public synchronized void produce(int i) {
+        while (deque.size() >= 10 ) {
```
[the remark this answers is not in the document]
Comment (VictPol) [reply src/main/java/com/rakovets/course/java/core/practice/concurrency_thread_synchronization/project_producer_consumer/task_02/Store.java:15]: fixed, parameter added to constructor

File: src/main/java/com/rakovets/course/java/core/practice/concurrency_thread_synchronization/project_sky_net/task_01/Demo.java (hunk 2 above)
Comment (VictPol) [reply src/main/java/com/rakovets/course/java/core/practice/concurrency_thread_synchronization/project_sky_net/task_01/Demo.java]: This project Skynet has been completly renewed, all files are new

File: src/main/java/com/rakovets/course/java/core/practice/concurrency_thread_synchronization/project_producer_consumer/task_05/ConsumerThread.java (hunk 1 above)
Comment (VictPol) [reply src/main/java/com/rakovets/course/java/core/practice/concurrency_thread_synchronization/project_producer_consumer/task_05/ConsumerThread.java]: fixed

**Pull request closed without merging**